src/front/glsl/parser.rs

```diff
@@ -213,20 +213,35 @@ pomelo! {
 
     postfix_expression ::= primary_expression;
     postfix_expression ::= postfix_expression LeftBracket integer_expression RightBracket {
-        // TODO
+        //TODO
         return Err(ErrorKind::NotImplemented("[]"))
     }
     postfix_expression ::= function_call;
-    postfix_expression ::= postfix_expression Dot FieldSelection {
-        // TODO
-        return Err(ErrorKind::NotImplemented(".field"))
+    postfix_expression ::= postfix_expression(e) Dot Identifier(i) /* FieldSelection in spec */ {
+        let type_inner = extra.resolve_type(e.expression)?;
+        if let TypeInner::Struct{members} = type_inner {
+            let index = members.iter().position(|m| m.name == Some(i.1.clone()));
+            if let Some(index) = index {
```

Comment:
you can shorten this by `let index = members.(...).ok_or(ErrorKind::UnknownField(i.0, i.1))?;`

Reply:
done